src/implementations/sign/sign.ts

```diff
@@ -0,0 +1,73 @@
+import { dirSync } from "tmp";
+import mkdirp from "mkdirp";
+import path from "path";
+import { documentsInDirectory, readDocumentFile, writeDocumentToDisk } from "../utils/disk";
+import { sign as oaSign, ProofType, SignedWrappedDocument } from "@govtechsg/open-attestation";
+
+export enum Output {
+  File,
+  Directory,
+  StdOut
+}
+
+interface SignArguments {
+  rawDocumentsPath: string;
+  outputPath?: string;
+  outputPathType: Output;
+  privateKey: string;
+  publicKey: string;
+}
+
+export const sign = async ({
+  rawDocumentsPath,
+  outputPath,
+  outputPathType,
+  privateKey,
+  publicKey
+}: SignArguments): Promise<SignedWrappedDocument<any>[]> => {
+  if (outputPath) {
+    // Create output dir
+    mkdirp.sync(outputPathType === Output.File ? path.parse(outputPath).dir : outputPath);
+  }
+
+  // Create intermediate dir
+  const { name: intermediateDir, removeCallback } = dirSync({
+    unsafeCleanup: true
+  });
+
+  // Copy to intermediate dir
+  const documentFileNames = await documentsInDirectory(rawDocumentsPath);
```

Comment:
why copying to an intermediate directory ? read the file, sign and write the the output ?